pkg/api/api.go

```diff
@@ -374,6 +374,7 @@ func (a *API) RegisterQueryAPI(handler http.Handler) {
 	a.RegisterRoute(path.Join(a.cfg.PrometheusHTTPPrefix, "/api/v1/label/{name}/values"), handler, true, "GET")
 	a.RegisterRoute(path.Join(a.cfg.PrometheusHTTPPrefix, "/api/v1/series"), handler, true, "GET", "POST", "DELETE")
 	a.RegisterRoute(path.Join(a.cfg.PrometheusHTTPPrefix, "/api/v1/metadata"), handler, true, "GET")
+	a.RegisterRoute(path.Join(a.cfg.PrometheusHTTPPrefix, "/api/v1/status/buildinfo"), handler, true, "GET")
```

Comment:
Honest question, do we need `auth` to be true for this API? I don't imagine the result of this API is tenant specific right?

Reply:
I don't think auth means we need this api to be tenant specific. Just need to make sure tenants can access it

Reply:
> Just need to only valid tenants can access the api

I guess that's what I meant by "tenant specific", do we want the user to pass in `X-Scope-OrgID` to use this API?

Reply:
Yeah I got your point… I am not sure if exposing the Cortex version info has any security concern.
My initial idea is to at least has one layer of check to ensure users are valid to get this information.
I could make it non auth, too. What do you think?

Reply:
I gave it another thought while driving, I think let’s keep the auth for future extensibility. Maybe we can disable some feature for some tenants in the futures because of abuse or something.